runtime/doc/dev_vimpatch.txt

```diff
@@ -302,4 +302,40 @@ used in new documentation:
 - `{Only when compiled with ...}`: the vast majority of features have been
   made non-optional (see https://github.com/neovim/neovim/wiki/Introduction)
 
+==============================================================================
+FILETYPE DETECTION                 *dev-vimpatch-filetype*
+
+Nvim's [filetype detection](https://github.com/neovim/neovim/blob/master/runtime/lua/vim/filetype.lua)
```

Comment:
Nit: This is Markdown syntax, but this is a Vim help buffer. Probably fine, but I think we typically would put the URL on a separate line with a "See " prefix or something.

And since this is linking to a a file in Nvim's runtime we could just use a file path which will let people easily jump to the file inside of Nvim with `gf`.

```suggestion
See $VIMRUNTIME/runtime/lua/vim/filetype.lua
```

Reply:
Addressed f47dae74398c2db981c8266494c2035953227489 with `$VIMRUNTIME/lua/vim/filetype.lua` path.